pkgs/by-name/sw/swaybg/package.nix

```diff
@@ -1,7 +1,7 @@
 { lib, stdenv, fetchFromGitHub
 , meson, ninja, pkg-config, scdoc
 , wayland, wayland-protocols, cairo, gdk-pixbuf
-, wayland-scanner
+, wayland-scanner, librsvg, gnome, wrapGAppsHook
```

Comment:
```suggestion
, wayland-scanner, wrapGAppsHook4
```

Reply:
Is it a gtk4 app?

Reply:
For gdk-pixbuf, `wrapGAppsNoGuiHook` should be enough.

Reply:
Thanks.

What about swaylock?

Reply:
from the nixpkgs manual:
> wrapGAppsNoGuiHook for programs without a graphical interface. Same as the above but does not bring gtk3 and librsvg into the closure.

I tested `wrapGAppsNoGuiHook` with an overlay and it does not work.
(Adding `librsvg` to  nativeBuildInputs does not solve this for me)

Reply:
`librsvg` would probably need to go to `buildInputs`.

Reply:
```nix
prev.swaylock.overrideAttrs (old: {
      preFixup = ''
        gappsWrapperArgs+=(
          --set GDK_PIXBUF_MODULE_FILE ${final.librsvg}/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache
        )
      '';
      nativeBuildInputs = old.nativeBuildInputs ++ [final.wrapGAppsNoGuiHook];
    });
```

This works for me. (Inspired by https://github.com/NixOS/nixpkgs/blob/5048836ab7a6de47e490a6e6259b15822d5f5753/pkgs/applications/audio/hushboard/default.nix)

Reply:
GDK_PIXBUF_MODULE_FILE is already included by the hook:

https://github.com/NixOS/nixpkgs/blob/831ec770471104081922bf82ab077162ea22c6d9/pkgs/build-support/setup-hooks/wrap-gapps-hook/wrap-gapps-hook.sh#L14

Reply:
🤦Yes, adding `librsvg` to `nativeBuildInputs` and `buildInputs` and adding `wrapGAppsNoGuiHook` to `nativeBuildInputs` works for me.

Reply:
Adding librsvg was not necessary for me, could you try without?

Reply:
Removing `librsvg` from either `nativeBuildInputs` or `buildInputs` results in an white lockscreen (only swaylock) and an error (both swaylock and swaybg): 
```
[background-image.c:30] Failed to load background image (Couldn’t recognize the image file format for file ...
```
for me.

Reply:
Huh that's interesting because it does work for me in swaylock and I use the state which I initially opened this PR in.

Reply:
The initial state worked for me too, but after switching to `wrapGAppsNoGuiHook` it requires `librsvg` for me.

Swaylock already worked with SVGs when started from my terminal but did not work when started from systemd (swayidle).
To test it, I stopped swayidle and used:
```bash
systemd-run --user --pty --property=Environment="WAYLAND_DISPLAY=wayland-1" swaylock --config none --image=/nix/store/vyay3ccbrjall7qv31r30h46d8wyjmky-gnome-backgrounds-47.0/share/backgrounds/gnome/blobs-d.svg
```